### PR TITLE
Allow `selected_index=none` in selection container widgets

### PR DESCRIPTION
**Pull request by pbugnion**
Prior to this commit, the way to *unset* the selection for a tab or accordion widget was to set `selected_index = -1`. Having to set it to `None` would be better (see issue #1493 ). 

This PR allows setting the index to None. The bulk of the changes are in the `Selection` class, which now allows setting the index to null.

~~This is what's left to do~~:
 - [x] explicitly disable setting selected_index to an out of range value
 - [x] unit tests for Selection class 
 - [x] amend docs

This is now ready for review.

### CODE REVIEW

**Comment by pbugnion** [comment]
screen shot 2017-07-11 at 07 30 17

File: packages/controls/src/phosphor/tabpanel.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ class TabPanel extends Widget {
    * Set the index of the currently selected tab.
    *
    * #### Notes
-   * If the index is out of range, it will be set to `-1`.
+   * If the index is out of range, it will be set to `null`.
```
Comment (jasongrout) [inline packages/controls/src/phosphor/tabpanel.ts]: It looks like the logic here is still missing for out of range values.

File: packages/controls/src/phosphor/tabpanel.ts [hunk 2]
```diff
@@ -120,7 +120,7 @@ class TabPanel extends Widget {
    * Get the index of the currently selected tab.
    *
    * #### Notes
-   * This will be `-1` if no tab is selected.
+   * This will be `null` if no tab is selected.
    */
   get currentIndex(): number {
     return this.tabBar.currentIndex;
```
Comment (jasongrout) [inline packages/controls/src/phosphor/tabpanel.ts]: Logic needs to be added here to return null if the tab bar index is -1.

File: packages/controls/src/phosphor/tabpanel.ts (hunk 1 above)
Comment (jasongrout) [inline packages/controls/src/phosphor/tabpanel.ts]: Actually, you don't need to do anything here - the tab bar handles the logic to set its current index to -1 if out of range. We'll just handle the translation to null up where we retrieve the index.

**Comment by jasongrout** [comment]
Thanks!

**Pull request merged**